application/src/main/scala/com/gu/deskcomexporttool/DeskComClient.scala

```diff
@@ -29,55 +29,64 @@ object DeskComClient {
 
     implicit val interactionLinkDecoder: Decoder[Link] = deriveDecoder(derivation.renaming.snakeCase)
     implicit val interactionLinksDecoder: Decoder[InteractionLinks] = deriveDecoder(derivation.renaming.snakeCase)
+    implicit val interactionResultsLinksDecoder: Decoder[InteractionResultsLinks] = deriveDecoder(derivation.renaming.snakeCase)
     implicit val interactionDecoder: Decoder[Interaction] = deriveDecoder(derivation.renaming.snakeCase)
     implicit val interactionEmbeddedDecoder: Decoder[GetInteractionsEmbedded] = deriveDecoder(derivation.renaming.snakeCase)
     implicit val interactionResponseDecoder: Decoder[GetInteractionsResponse] = deriveDecoder(derivation.renaming.snakeCase)
 
-    override def getAllInteractions(page: Int, pageSize: Int): EitherT[Future, DeskComApiError, List[Interaction]] = {
+    override def getAllInteractions(sinceId: String, pageSize: Int): EitherT[Future, DeskComApiError, GetInteractionsResponse] = {
       for {
         httpResponse <- httpClient.request(
           HttpRequest(
             "GET",
-            s"${config.baseUrl}/api/v2/interactions?page=$page&per_page=$pageSize&sort_field=created_at&sort_direction=asc",
+            s"${config.baseUrl}/api/v2/interactions?since_id=$sinceId&per_page=$pageSize",
             List(authHeader)
           )
-        ).leftMap(httpError => DeskComApiError(s"Request for interactions failed: $httpError"))
-        _ <- EitherT.fromEither(validateStatusCode(httpResponse.statusCode))
+        ).leftMap(httpError => DeskComUnexpectedApiError(s"Request for interactions failed: $httpError"))
+        _ <- EitherT.fromEither(validateStatusCode(httpResponse))
         parsedResponseBody <- EitherT.fromEither(parseGetAllInteractions(httpResponse.body))
-      } yield parsedResponseBody._embedded.entries
+      } yield parsedResponseBody
     }
 
     private def parseGetAllInteractions(body: String): Either[DeskComApiError, GetInteractionsResponse] = {
       decode[GetInteractionsResponse](body)
         .leftMap { parsingFailure =>
-          log.debug(s"Failed to parse response error:$parsingFailure response: $body")
-          DeskComApiError(s"Failed to parse interaction response: $parsingFailure")
+          log.debug(s"Failed to parse response response: $body error:$parsingFailure")
+          DeskComUnexpectedApiError(s"Failed to parse interaction response: $parsingFailure")
         }
     }
 
-    private def validateStatusCode(statusCode: Int): Either[DeskComApiError, Unit] = {
-      statusCode match {
+    private def validateStatusCode(httpResponse: HttpResponse): Either[DeskComApiError, Unit] = {
+      httpResponse.statusCode match {
         case 200 => Right(())
-        case statusCode => Left(DeskComApiError(s"Interactions endpoint returned status: $statusCode"))
+        case 422 => Left(DeskComUnprocessableEntity(httpResponse.body.take(1000)))
```

Comment:
What is the magic 1000?

Reply:
these are just truncating bodies for log messages, ill make it more explicit